examples/mnist/main.py

```diff
@@ -48,23 +57,33 @@ def outdef(self):
                 )
         return [y_def]
     
+    def distill(self, x):
+        x = x.view(-1, 1, 28, 28)
+        x = torch.tanh(self.dist_conv1(x))
+        x = self.dist_average1(x)
+        x = torch.tanh(self.dist_conv2(x))
+        x = self.dist_average2(x)
+        x = torch.tanh(self.dist_conv3(x))
+        x = x.view(-1, x.shape[1])
+        x = F.relu(self.dist_fc1(x))
+        x = F.relu(self.dist_fc2(x))
+        return x
```

Comment:
Is the distill always going to look like the fwd pass?

Reply:
Not always, but in this case the output of the network is [batch_size, 10] to fit mnist. In all likelihood, outputs would be 1024 + embeddings in which case, the distillation layer would learn x -> 1024